ads_plugin.py

```diff
@@ -0,0 +1,190 @@
+
+class Ads_control_v2(ctrlpanel.ControlPanel):
+
+    def __init__(self, *args, **kwargs):
+        ctrlpanel.ControlPanel.__init__(self, *args, **kwargs)
+
+        # Adding sizers to the control panel
+        sizerH = wx.BoxSizer(wx.HORIZONTAL)
+        sizerV1 = wx.BoxSizer(wx.VERTICAL)
+        sizerV2 = wx.BoxSizer(wx.VERTICAL)
+        sizerV3 = wx.BoxSizer(wx.VERTICAL)
+        sizerV4 = wx.BoxSizer(wx.VERTICAL)
+
+        sizerH.Add(sizerV1, 1, wx.EXPAND | wx.LEFT | wx.RIGHT | wx.TOP | wx.BOTTOM, border=10)
+        sizerH.Add(sizerV2, 1, wx.EXPAND | wx.LEFT | wx.RIGHT | wx.TOP | wx.BOTTOM, border=10)
+        sizerH.Add(sizerV3, 1, wx.EXPAND | wx.LEFT | wx.RIGHT | wx.TOP | wx.BOTTOM, border=10)
+        sizerH.Add(sizerV4, 1, wx.EXPAND | wx.LEFT | wx.RIGHT | wx.TOP | wx.BOTTOM, border=10)
+
+        # Adding widgets to the control panel
+
+        ADS_logo = self.getLogo()
+        sizerV1.Add(ADS_logo, flag=wx.SHAPED, proportion=1)
+
+
+        citationBox = wx.TextCtrl(self, value=self.getCitation(), size=(100, 50), style=wx.TE_MULTILINE)
+        sizerV1.Add(citationBox, flag=wx.SHAPED, proportion=1)
+
+
+        loadPng_button = wx.Button(self, label = "Load PNG file")
+        loadPng_button.Bind(wx.EVT_BUTTON, self.onLoadPngButton)
+        sizerV2.Add(loadPng_button, flag=wx.SHAPED, proportion=1)
+
+        saveSegmentation_button = wx.Button(self, label="Save segmentation")
+        sizerV2.Add(saveSegmentation_button, flag=wx.SHAPED, proportion=1)
+
+        self.modelChoices = ['SEM', 'TEM', 'other']
+        self.modelCombobox = wx.ComboBox(self, choices=self.modelChoices, size=(100, 20), value='Select the modality')
+        sizerV3.Add(self.modelCombobox, flag=wx.SHAPED, proportion=1)
+
+        applyModel_button = wx.Button(self, label = 'Apply ADS prediction model')
+        applyModel_button.Bind(wx.EVT_BUTTON, self.onApplyModel_button)
+        sizerV3.Add(applyModel_button, flag=wx.SHAPED, proportion=1)
+
+        fillMyelin_button = wx.Button(self, label='Fill closed myelin')
+        sizerV4.Add(fillMyelin_button, flag=wx.SHAPED, proportion=1)
+
+        self.SetSizer(sizerH)
+
+        self.imageDirPath = None
+
+
+
+    def onLoadPngButton(self, event):
+        # Ask the user which file he wants to convert
+        with wx.FileDialog(self, "select PNG file", wildcard="PNG files (*.png)|*.png",
+                           style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST) as fileDialog:
+
+            if fileDialog.ShowModal() == wx.ID_CANCEL:
+                return
+
+            inFile = fileDialog.GetPath()
+
+
+        if inFile[-4:] != '.png':
+            print('This is not a png file')
+            return
+
+        # Store the directory path and image name for later use
+        self.imageDirPath = os.path.dirname(inFile)
+        self.pngImageName = inFile[self.imageDirPath.__len__()+1:]
+
+        self.loadPngImageFromPath(inFile)
+
+
+
+    def onApplyModel_button(self, event):
+
```

Comment:
there is a lot of empty lines. Please conform to PEP8 coding rules.
also add docstrings at the beginning of each function. See good example here https://thomas-cokelaer.info/tutorials/sphinx/docstring_python.html#template-py-source-file

Reply:
I applied the formatting from the [black](https://github.com/python/black) package to this file, which adheres pretty close to PEP8, so I think this comment can be resolved.